rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -63,6 +76,37 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     protected abstract String[] splitCSV(String barcodeCSV);
 
+    protected void populateBarcodeData(Intent data) throws JSONException {
+
+        JSONObject jsonObject = new JSONObject(data.getStringExtra("data"));
+
+        formUtils.getNextUniqueIds(null, (args, uniqueIds) -> {
+            List<String> ids = new ArrayList<>();
+            for (UniqueId uniqueId : uniqueIds) {
+                String currUniqueId = uniqueId.getOpenmrsId();
+                if (StringUtils.isNotBlank(currUniqueId)) {
+                    currUniqueId = currUniqueId.replace("-", "");
+                    ids.add(currUniqueId);
+                }
+            }
```

Comment:
This code is duplicated https://github.com/onaio/rdt-standard/blob/5dd73bbd960e15ba6553796276d8308d3b9e7566/rdt-open-reader/src/main/java/io/ona/rdt/util/FormLauncher.java#L62. Could we refactor this into a util that converts UniqueIds to String ids?

Reply:
updated.